src/utilities/utilities.ts

```diff
@@ -173,3 +176,27 @@ export async function getXCTestPath(): Promise<string> {
     }
     return path.join(platformPath, "Developer", "Library", `XCTest-${version}`, "usr", "bin");
 }
+
+/**
+ * @returns SwiftPM flag for enabling test discovery
+ */
+export async function testDiscoveryFlag(ctx: FolderContext): Promise<string[]> {
+    // Test discovery is only available in SwiftPM 5.1 and later.
+    if (ctx.workspaceContext.swiftVersion.isLessThan(new Version(5, 1, 0))) {
+        return [];
+    }
+    // Test discovery is always enabled on Darwin.
+    if (process.platform !== "darwin" && ctx.swiftPackage.getTargets("test").length > 0) {
+        const alwaysDiscoverTests = vscode.workspace
+            .getConfiguration("swiftpm")
+            .get<boolean>("testDiscovery.always", true);
+        const hasLinuxMain = await pathExists(ctx.folder.fsPath, "Tests", "LinuxMain.swift");
```

Comment:
Instead of testing for the existence of `LinuxMain.swift` all the time, perhaps the existence of it should be stored in SwiftPackage and setup in `SwiftPackage.create`